src/components/account/StakingRecords.tsx

```diff
@@ -0,0 +1,130 @@
+import { Card, Tabs } from 'antd';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import type { Option } from '@polkadot/types';
+import { StakingRecordType, UnbondDataSourceState, UnbondType } from '../staking/interface';
+import { useApi, useBestNumber, useStaking } from '../../hooks';
+import { CustomTab } from '../widget/CustomTab';
+import type { DarwiniaStakingStructsStakingLedger } from '../../api-derive/types';
+import { UnbondRecords } from './UnbondRecords';
+import { LockedRecords } from './LockedRecords';
+
+export function StakingRecords() {
+  const { api, network } = useApi();
+  const { controllerAccount } = useStaking();
+  const { bestNumber } = useBestNumber();
+  const { t } = useTranslation();
+  const [activeKey, setActiveKey] = useState(StakingRecordType.LOCKS);
+  const [ledger, setLedger] = useState<DarwiniaStakingStructsStakingLedger | null>();
+  const [unbondedDataSource, setUnondedDataSource] = useState<UnbondDataSourceState[]>([]);
+  const [unbondingDataSource, setUnondingDataSource] = useState<UnbondDataSourceState[]>([]);
+
+  useEffect(() => {
+    if (!controllerAccount) {
+      setLedger(null);
+      return;
+    }
+
+    let unsub: () => void = () => undefined;
+
+    (async () => {
+      unsub = await api.query.staking.ledger(controllerAccount, (res: Option<DarwiniaStakingStructsStakingLedger>) => {
+        setLedger(res.isSome ? res.unwrap() : null);
+      });
+    })();
+
+    return () => unsub();
+  }, [api, controllerAccount]);
+
+  useEffect(() => {
+    const { ringUnbondeds, ringUnbondings } = ledger?.ringStakingLock.unbondings.reduce(
+      ({ ringUnbondeds, ringUnbondings }, { amount, until }) => {
+        if (until.ltn(bestNumber || 0)) {
+          ringUnbondeds.push({
+            amount,
+            until,
+            status: UnbondType.UNBONDED,
+            symbol: network.tokens.ring.symbol,
+          });
+        } else {
+          ringUnbondings.push({
+            amount,
+            until,
+            status: UnbondType.UNBONDING,
+            symbol: network.tokens.ring.symbol,
+          });
+        }
+        return { ringUnbondeds, ringUnbondings };
+      },
+      { ringUnbondeds: [], ringUnbondings: [] } as {
+        ringUnbondeds: UnbondDataSourceState[];
+        ringUnbondings: UnbondDataSourceState[];
+      }
+    ) || { ringUnbondeds: [], ringUnbondings: [] };
```

Comment:
DRY, the calculation logic of kton is the same as the ring processess, too many code here

Reply:
It's different. The respective `status` and `symbol` are different, there is no good way to combine